### ISSUE TITLE
AddSignIn and AddProtectedWebApi IServiceCollection extensions should return AuthenticationBuilder

### ISSUE THREAD

**Issue by pmaytak** [AzureAD/microsoft-identity-web#90]
Currently there are a few `IServiceCollection` extension methods that create and update an `AuthenticationBuilder` object but return an `IServiceCollection` object.
Methods in focus are:
- two `AddProtectedWebApi` methods (with [config name](https://github.com/AzureAD/microsoft-identity-web/blob/d7570bb05ad0576b3595b2def64de883a2088326/src/Microsoft.Identity.Web/WebApiServiceCollectionExtensions.cs#L34) and [config actions](https://github.com/AzureAD/microsoft-identity-web/blob/d7570bb05ad0576b3595b2def64de883a2088326/src/Microsoft.Identity.Web/WebApiServiceCollectionExtensions.cs#L64)) in `WebApiServiceCollectionExtensions`
- two `AddSignIn` overloads (with [config name](https://github.com/AzureAD/microsoft-identity-web/blob/d7570bb05ad0576b3595b2def64de883a2088326/src/Microsoft.Identity.Web/WebAppServiceCollectionExtensions.cs#L41) and [config actions](https://github.com/AzureAD/microsoft-identity-web/blob/d7570bb05ad0576b3595b2def64de883a2088326/src/Microsoft.Identity.Web/WebAppServiceCollectionExtensions.cs#L75)) in `WebAppServiceCollectionExtensions`

It seems that in the ASP.NET Core `IServiceCollection` extensions classes, if the extension method only adds some services to the services collection, then it's fine to return an `IServiceCollection` object, but if that method has more setup using a builder, then builder object is returned.

Today, by returning services, we enable (simple) code like:
```CSharp
services.AddSignIn(Configuration)
        .AddProtectedWebApiCallsWebAPi(Configuration)
        .AddTokenCacheXXX();
```
A builder has a `Services` references and by returning it, the same code can be:
```CSharp
services.AddSignIn(Configuration);
        .Services
        .AddProtectedWebApiCallsWebAPi(Configuration)
        .AddTokenCacheXXX();
```
`AuthenticaitonBuilder` has some specific authentication related methods that can be chained to add functionality to the builder.

The question - do we return a more specific `AuthenticationBuilder` object or a more general `IServicesCollection` object.

**Comment by Tratcher** [comment]
This should go a step further and move the AddSignIn method from IServiceCollection to AuthentiationBuilder. All other auth providers chain off of the AuthenticationBuilder rather than the service collection. If you need to manipulate services, those are available from the AuthenticationBuilder.

Also note that an auth provider should not be calling AddAuthenticaiton itself to specify the app default schemes. This breaks composition of different auth providers. AddAuthentication should should called directly in samples and templates.

**Comment by jennyf19** [comment]
Included in [0.2.0-preview release](https://github.com/AzureAD/microsoft-identity-web/releases/tag/0.2.0-preview)
